servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/ServiceTalkHttp2FrameLogger.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright © 2020 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.http.netty;
+
+import io.servicetalk.logging.api.FixedLevelLogger;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufUtil;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.http2.Http2Flags;
+import io.netty.handler.codec.http2.Http2FrameLogger;
+import io.netty.handler.codec.http2.Http2Headers;
+import io.netty.handler.codec.http2.Http2Settings;
+import io.netty.handler.logging.LogLevel;
+
+final class ServiceTalkHttp2FrameLogger extends Http2FrameLogger {
+    private static final int BUFFER_LENGTH_THRESHOLD = 64;
+    private final FixedLevelLogger logger;
+    private final boolean logUserData;
+
+    ServiceTalkHttp2FrameLogger(final FixedLevelLogger logger, final boolean logUserData) {
```

Comment:
Will be great to enhance `Http2FrameLogger` in netty to support "logUserData" option in future. Looks like a useful feature for all netty users.

Reply:
agreed this is a candidate to upstream